nalgebra-macros/src/lib.rs

```diff
@@ -322,3 +322,268 @@ pub fn point(stream: TokenStream) -> TokenStream {
     };
     proc_macro::TokenStream::from(output)
 }
+
+enum ConcatElem {
+    Zero,
+    One,
+    Expr(Expr),
+}
+
+impl ConcatElem {
+    fn from_expr(expr: Expr) -> Self {
+        match &expr {
+            Expr::Lit(syn::ExprLit { lit: Lit::Int(ilit), .. }) => {
+                if ilit.base10_digits() == "0" {
+                    return ConcatElem::Zero
+                } else if ilit.base10_digits() == "1" {
+                    return ConcatElem::One
+                }
+            }, 
+            _ => (),
+        };
+        ConcatElem::Expr(expr)
+    }
+
+    fn is_expr(&self) -> bool {
+        match self {
+            ConcatElem::Zero => false,
+            ConcatElem::One => false,
+            ConcatElem::Expr(_) => true
+        }
+    }
+}
+
+/// Constructs a new matrix by concatenating the given matrices.
+///
+/// **Note: Requires the `macros` feature to be enabled (enabled by default)**.
+///
+/// The syntax is similar to the [`matrix!`](./macro.matrix.html) and
+/// [`dmatrix`](./macro.dmatrix.html) macros. However the elements should
```

Comment:
It's better to use an [intra-doc link](https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html) rather than directly linking to `html` pages.

Reply:
Nice! Was this feature recently introduced? I don't think you could do this the last time I looked at how to link to other doc-pages.

Reply:
[Since 1.48](https://blog.rust-lang.org/2020/11/19/Rust-1.48.html)!